cmd/monaco/generate/dependencygraph/dependencygraph.go

```diff
@@ -86,7 +87,20 @@ func writeGraphFiles(fs afero.Fs, manifestPath string, environmentNames []string
 		}
 	}
 
-	graphs := graph.New(projects, m.Environments.Names())
+	var opts []graph.NodeOption
+	if writeJSONIDs {
+		log.Debug("Encoding DOT Node IDs as JSON")
+		opts = append(opts, func(n *graph.ConfigNode) {
+			s, err := json.Marshal(n.Config.Coordinate)
+			if err == nil {
+				n.DOTEncoding = string(s)
+			} else {
+				n.DOTEncoding = fmt.Sprintf(`{ "encodingError": "%v" }`, err)
```

Comment:
Perhaps it is better to use `json.Marshal` to encode the encoding error?

Reply:
adapted as discussed internally - encoding will be an empty JSON object so we don't break any automation reading the graph and an error is logged.
In general we can still assume that this is a 'should never happen' case, as the coordinate is a simple struct of strings that should always be possible to marshal - and any substantial modification breaking marshalling, would also have impact on most of monaco.